go-controller/pkg/ovn/network_controller_info_pods.go

```diff
@@ -0,0 +1,694 @@
+package ovn
```

Comment:
should this file and network_controller_info.go be in a subdirectory package? Like

pkg/ovn/network-controller-info/
...info.go
...pods.go

or maybe pkg/ovn/network-controller-shared/

wdyt?

Reply:
@trozet @jcaamano Not sure if I understand this comment correctly. But it seems to me that would be a lot of changes: We'd need to change a lot of baseNetworkController/NetworkControllerInfo fields to be capitalized string, and also needs to move quite some existing ovn package functions/structures into the new package, e.g., namespaceInfo , so they don't have import cycle. But because the current code is so tightly connected, both ovn/networkControllerInfo packages have functions to directly access the fields of these structures directly (so all would need to be capitalized as well).

Because of this, I am not sure that making that into a separate package is a good idea.

Reply:
The `BaseNetworkController` shared methods would need to be exported. But as for the fields, in a perfect world BaseNetworkController fields would not be accessed from outside BaseNetworkController (as that's why we are sharing functionality through its methods), and I guess BaseNetworkController would not need to access fields outside of it.

As for other common shared utilities, those would need to be moved to a third package as you say.

If it is too much work, I guess we can do it later. But I hoped this would be of the things we would achieve with this effort. Maybe not in this PR but not much later.

Anyway, I will let @trozet express his opinion here.

Reply:
agree. to make it subfolder package, ovn should not directly access any field of BaseNetworkController , but only through exported interfaces/functions. I'd rather defer this to a later work.

Reply:
we can defer it until later.